aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -53,19 +55,46 @@ void expm1_kernel_cuda(TensorIteratorBase& iter) {
 
 // We manually overload rsqrt because std::rsqrt does not work with complex types.
 template<typename scalar_t>
-__host__ __device__ static inline scalar_t rsqrt_wrapper(scalar_t v) {
+C10_HOST_DEVICE static inline scalar_t rsqrt_wrapper(scalar_t v) {
   return ::rsqrt(v);
 }
 
 template<typename T>
-__host__ __device__ static inline c10::complex<T> rsqrt_wrapper(c10::complex<T> v) {
+C10_HOST_DEVICE static inline c10::complex<T> rsqrt_wrapper(c10::complex<T> v) {
   const c10::complex<T> one = c10::complex<T>(1.0, 0);
   // std::sqrt for c10::complex is overloaded in c10/util/complex_math.h
   return one / ::sqrt(v);
 }
 
+const char rsqrt_name[] = "rsqrt_kernel";
 void rsqrt_kernel_cuda(TensorIteratorBase& iter) {
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(
+  auto common_dtype = iter.common_dtype();
+  if (at::isComplexType(common_dtype)) {
+    #if AT_USE_JITERATOR()
+      static const auto rsqrt_string = jiterator_stringify(
+          template <typename T>
+          T rsqrt_kernel(T x) {
+            const T one = T{1};
+            // std::sqrt for c10::complex is overloaded in c10/util/complex_math.h
```

Comment:
Shouldn't the complex headers the jiterator includes define `std::sqrt`? I thought that file worked by extending the standard namespace with the complex math operations

cc @zasdfgbnm

Reply:
It should: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/cuda/llvm_complex.cpp#L727

Reply:
I think it worked out due to argument dependent lookup.

Have qualified the call with `std` namespace.